h5py/tests/old/test_h5pl.py

```diff
@@ -0,0 +1,56 @@
+# This file is part of h5py, a Python interface to the HDF5 library.
+#
+# http://www.h5py.org
+#
+# Copyright 2008-2019 Andrew Collette and contributors
+#
+# License:  Standard 3-clause BSD; see "license.txt" for full license terms
+#           and contributor agreement.
+
+from __future__ import absolute_import
+
+try:
+    import unittest2 as ut
+except ImportError:
+    import unittest as ut
+
+import h5py
+from h5py import h5pl
+
+from ..common import TestCase
+
+@ut.skip('The tests have side effects')
+@ut.skipUnless(h5py.version.hdf5_version_tuple >= (1, 10, 1), 'HDF5 1.10.1+ required')
+class TestSearchPaths(TestCase):
+
+    def test_default(self):
+        self.assertEqual(h5pl.size(), 1)
```

Comment:
No, quite the contrary: it assumes the default value (see next line).
Running in the separate process, without side effects and predictable `HDF5_PLUGIN_PATH` sounds good. What's the most elegant way to do so?

Reply:
I don't know if pytest has some neat magic to make this convenient, but I would do something like this:

```python
from subprocess import run, PIPE, STDOUT

res = run([sys.executable, '-m', 'pytest', '...'], stdout=PIPE, stderr=STDOUT)
print(res.stdout)
assert res.returncode == 0
```

The `...` there would contain whatever arguments are needed to make pytest run the tests in a module it would otherwise skip.

Reply:
PR #1003 was merged, so as far as I know, it uses pytest.

A lot of tests are still written with the unittest API; pytest can find and run these with no problem. I'm not personally enthusiastic about 'converting' existing tests to a more idiomatic pytest style - it sounds like a lot of change for very little benefit - but I'd say it's OK to write new tests assuming pytest.

Reply:
What do you think about the approach of https://github.com/pytest-dev/pytest-forked?

Reply:
Not enthusiastic! It looks like it's implemented as a command-line option which applies to all tests, whereas we want to apply this to only a specific set of tests that manipulate global state. Also, fork-without-exec is easy to mess up.

I can imagine this being a useful general utility, though - run some tests in a subprocess and integrate their results back into the main test results. Pytest itself has some machinery for testing pytest plugins which might be a useful starting point for something like this: https://docs.pytest.org/en/latest/writing_plugins.html#testing-plugins